src/Umbraco.Web.UI.Client/src/views/content/content.create.controller.js

```diff
@@ -73,25 +74,24 @@ function contentCreateController($scope,
     }
 
     function createOrSelectBlueprintIfAny(docType) {
-        // map the blueprints into a collection that's sortable in the view
-        var blueprints = _.map(_.pairs(docType.blueprints || {}), function (pair) {
-            return {
-                id: pair[0],
-                name: pair[1]
-            };
-        });
         $scope.docType = docType;
-        if (blueprints.length) {
-            if (blueprintConfig.skipSelect) {
-                createFromBlueprint(blueprints[0].id);
-            } else {
-                $scope.selectContentType = false;
-                $scope.selectBlueprint = true;
-                $scope.selectableBlueprints = blueprints;
-            }
-        } else {
+
+        const keys = Object.keys(docType.blueprints);
```

Comment:
You removed a null propagation.

Reply:
Yup, it wasn't required - blueprints will be an empty object if none exists for the doctype.

I think the failing test is related to using q to resolve the API requests and/or destructing that response. Test fails when allowedTypes is null, so looks to me like that test case is being asserted before the promises are resolved.

Reply:
Ah. Need $httpBackend.$flush() and/or $rootScope.$appy()?
Otherwise might be that the test lacks a "done" parameter to do the "old-style" async wait in the test.
Haven't got time to dig in, just glanced at the code from the phone. ☺